react-client/src/components/trade.jsx

```diff
@@ -1,33 +1,283 @@
+      }
+    } else if (this.props.trade.status === true
+      && moment(this.props.trade.time).fromNow() !== 'EXPIRED') {
+      this.setState({ liStatus: 'acceptedTrade' });
+      if (this.state.truthy) {
+        this.setState({
+          message: `You accepted ${this.props.trade.username2}'s trade for 
+          ${this.props.trade.food_id2} with your ${this.props.trade.food_id1} 
+          on ${moment(this.props.trade.time).format('ddd, MMM DD')} at 
+          ${moment(this.props.trade.time).format('h:mm a')} and have 
+          ${moment(this.props.trade.time).fromNow()} left to swap.`,
+        });
+      } else {
+        this.setState({
+          message: `${this.props.trade.username1} accepted your trade for his 
+          ${this.props.trade.food_id1} with your ${this.props.trade.food_id2} 
+          on ${moment(this.props.trade.time).format('ddd, MMM DD')} at 
+          ${moment(this.props.trade.time).format('h:mm a')} and have 
+          ${moment(this.props.trade.time).fromNow()} left to swap.`,
+        });
+      }
+    } else if (this.props.trade.status === true) {
+      this.setState({ liStatus: 'succssfulTrade' });
+      this.setState({
+        message: `You completed your trade of ${this.props.trade.food_id1} for 
+        ${this.props.trade.username2}'s ${this.props.trade.food_id2} on 
+        ${moment(this.props.trade.time).format('ddd, MMM DD')}.`,
+      });
+      this.setState({
+        buttons:
+  <div>
+    <button>Rate Trade</button>
+    <button>Trade Did Not Happen</button>
+  </div>,
+      });
+    } else {
+      this.setState({ liStatus: 'rejectedTrade' });
+      if (this.state.truthy) {
+        this.setState({
+          message: `You  rejected ${this.props.trade.username2}'s trade of 
+          ${this.props.trade.food_id2} for your ${this.props.trade.food_id1}.`,
+        });
+      } else {
+        this.setState({
+          message: `${this.props.trade.username1} reject your trade of 
+          ${this.props.trade.food_id2} for their ${this.props.trade.food_id1}.`,
+        });
+      }
+    }
+    this.setState({
+      firstStar: <UnRatedStar location={1} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
+      secondStar: <UnRatedStar location={2} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
+      thirdStar: <UnRatedStar location={3} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
+      fourthStar: <UnRatedStar location={4} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
+      fifthStar: <UnRatedStar location={5} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
+    });
+  }
+
+  updateStars(num) {
+    switch (num) {
+      case 1:
+        this.setState({
+          firstStar: <RatedStarLike location={1} click={this.updateStars} hover={this.hoverStars} conditional={this.state.clicked} />,
```

Comment:
Could you create a simple component in a function that only has location parameter?